### PR TITLE
Add Ubuntu support for exporters' reverse proxy

### PR DESCRIPTION
**Pull request by witekest**
Fixes: https://github.com/SUSE/spacewalk/issues/12548

### CODE REVIEW

File: prometheus-exporters-formula/prometheus-exporters/init.sls [hunk 1]
```diff
@@ -20,6 +27,13 @@ exporter_exporter:
       - pkg: exporter_exporter
     - watch_in:
       - service: exporter_exporter
+    - names:
+      - {{ exporters.exporter_exporter_service_config }}:
+        - source: {{ 'salt://prometheus-exporters/files/exporter-exporter-config.' ~ salt['grains.get']('os_family') }}
+  {% if grains['os_family'] == 'Debian' %}
+      - /etc/exporter_exporter.yaml:
+        - source: salt://prometheus-exporters/files/exporter-exporter.yaml
```
Comment (renner) [inline prometheus-exporters-formula/prometheus-exporters/init.sls:35]: What is this file used for, and why is it only needed for Debians?

**Review by renner — APPROVED** [review]
It looks good :+1: In general it would be great if we could avoid those `if grains['os_family'] = Debian` in the state by minimizing the differences between distributions, but I'm currently not sure if it is possible here.

File: prometheus-exporters-formula/prometheus-exporters/init.sls (hunk 1 above)
Comment (witekest) [reply prometheus-exporters-formula/prometheus-exporters/init.sls:35]: It's the central empty configuration file in [directory-based configuration setup](https://github.com/QubitProducts/exporter_exporter#directory-based-configuration). In SUSE we have it in [the package](https://build.opensuse.org/package/view_file/server:monitoring/golang-github-QubitProducts-exporter_exporter/exporter_exporter.yaml?expand=1). Debian upstream package does not include it. Instead it provides [configuration](https://salsa.debian.org/go-team/packages/prometheus-exporter-exporter/-/blob/debian/sid/expexp.yaml) for exporters we don't ship, so it has to be reconfigured anyway.

**Review by witekest — COMMENTED** [review]
We use upstream Debian package, so I would not like to make changes there. What we could do is to remove some configuration from the SUSE package and sync it with the Debian one.

**Comment by renner** [comment]
As discussed today there is no problem with merging it as it is, but we should aim to equalize reverse proxy packages long-term (at least after the Debian package is official) so we can keep the state simple and avoid special handling for certain distributions.

**Pull request merged**